### PR TITLE
TVL aggregation fix

### PR DESCRIPTION
**Pull request by bobo-k2**
TVL aggregation was completely broken. Some daily values were negative and for some days token price in USD was 0. I believe this happened when we merged #16, but not sure.
This is fixed now and TVL for the current day matches exactly with on chain data which gives me confidence that data for all days is correct.

image

### CODE REVIEW

File: src/mapping/tvl.ts [hunk 1]
```diff
@@ -28,60 +28,79 @@ export async function handleTvl(
...
   ctx: ProcessorContext<Store>,
-  initialDay: number
+  initialDay: number,
+  entities: Entities
 ): Promise<TvlAggregatedDaily> {
   // Pre-fetch all subperiods to determine if special handling is needed
   const subperiods = await ctx.store.find(Subperiod);
 
   let day = initialDay;
   while (true) {
```
Comment (impelcrypto) [inline src/mapping/tvl.ts:93]: how to break from the while loop?

File: src/utils/getUsdPrice.ts [hunk 2]
```diff
@@ -2,6 +2,10 @@ import https from "https";
 
 const priceCache = new Map<string, number>();
 
+function sleep(ms: number) {
```
Comment (impelcrypto) [inline src/utils/getUsdPrice.ts:5]:
```suggestion
function sleep(ms: number): Promise<void> {
```

File: src/mapping/tvl.ts (hunk 1 above)
Comment (gtg7784) [reply src/mapping/tvl.ts:93]: using return

**Review by gtg7784 — APPROVED** [review]
Looks good to me

**Review by impelcrypto — APPROVED** [review]

File: src/mapping/tvl.ts (hunk 1 above)
Comment (bobo-k2) [reply src/mapping/tvl.ts:93]: It can be undefined for first day of dapp staking but this is handled in `elseif` statement

**Pull request merged**